extensions/procfs/processors/ProcFsMonitor.cpp

```diff
@@ -0,0 +1,283 @@
+  gsl_Expects(session);
+  std::shared_ptr<core::FlowFile> flowFile = session->create();
+  if (!flowFile) {
+    logger_->log_error("Failed to create flowfile!");
+    yield();
+    return;
+  }
+
+  rapidjson::Document root = rapidjson::Document(rapidjson::kObjectType);
+  rapidjson::Value &body = prepareJSONBody(root);
+  auto current_cpu_stats = proc_fs_.getCpuStats();
+  auto current_disk_stats = proc_fs_.getDiskStats();
+  auto current_net_devs = proc_fs_.getNetDevs();
+  auto current_process_stats = proc_fs_.getProcessStats();
+
+  auto last_cpu_period = getAggregateCpuDiff(current_cpu_stats, last_cpu_stats_);
+
+  auto refresh_members = gsl::finally([&]{ refreshMembers(std::move(current_cpu_stats),
+                                                          std::move(current_disk_stats),
+                                                          std::move(current_net_devs),
+                                                          std::move(current_process_stats));});
+
+  processCPUInformation(current_cpu_stats, body, root.GetAllocator());
+  processDiskInformation(current_disk_stats, body, root.GetAllocator());
+  processNetworkInformation(current_net_devs, body, root.GetAllocator());
+  processProcessInformation(current_process_stats, last_cpu_period, body, root.GetAllocator());
+  processMemoryInformation(body, root.GetAllocator());
+
+  if (output_compactness_ == OutputCompactness::PRETTY) {
+    utils::PrettyJsonOutputCallback callback(std::move(root), decimal_places_);
+    session->write(flowFile, &callback);
+    session->transfer(flowFile, Success);
+  } else if (output_compactness_ == OutputCompactness::COMPACT) {
+    utils::JsonOutputCallback callback(std::move(root), decimal_places_);
+    session->write(flowFile, &callback);
+    session->transfer(flowFile, Success);
+  } else {
+    throw Exception(GENERAL_EXCEPTION, "Invalid output compactness");
+  }
+}
+
+rapidjson::Value& ProcFsMonitor::prepareJSONBody(rapidjson::Document& root) {
+  if (output_format_ == OutputFormat::OPENTELEMETRY) {
+      utils::OpenTelemetryLogDataModel::appendEventInformation(root, "PerformanceData");
+      utils::OpenTelemetryLogDataModel::appendHostInformation(root);
+      utils::OpenTelemetryLogDataModel::appendBody(root);
+      return root["Body"];
+  } else if (output_format_ == OutputFormat::JSON) {
+      return root;
+  } else {
+    throw Exception(GENERAL_EXCEPTION, "Invalid output format");
+  }
+}
+
+void ProcFsMonitor::setupDecimalPlacesFromProperties(const core::ProcessContext& context) {
+  std::string decimal_places_str;
+  if (!context.getProperty(DecimalPlaces.getName(), decimal_places_str) || decimal_places_str.empty()) {
+    decimal_places_ = std::nullopt;
+    return;
+  }
+
+  try {
+    uint32_t decimal_places = std::stoul(decimal_places_str);
+    if (decimal_places > std::numeric_limits<uint8_t>::max() || decimal_places == 0)
+      throw std::out_of_range("");
+    decimal_places_ = decimal_places;
```

Comment:
this will give unexpected results if `long` is 64 bits and `decimal_places_str` is something like 2^32 + 3, and although that's not going to happen, I would change this to
```suggestion
    const auto decimal_places = std::stoul(decimal_places_str);
    if (decimal_places > std::numeric_limits<uint8_t>::max() || decimal_places == 0) {
      throw Exception(PROCESS_SCHEDULE_EXCEPTION, "ProcFsMonitor Decimal Places property is zero or too large");
    }
    decimal_places_ = gsl::narrow<uint8_t>(decimal_places);
```